File: Ruby/bubble_sort.rb
def bubble_sort(arr)
    swapped = true
    counter = 0
    while swapped #so it doesn't continuously swap 
        swapped = false
        (0..arr.length - 2).each do |i|
            if arr[i] > arr[i+1]
                arr[i], arr[i+1] = arr[i+1], arr[i]
                swapped = true
            end
        end
    end
    arr
end

p bubble_sort([2,3,1,5,4])

File: JS/bubbleSort.js
const bubbleSort = (arr) => {
    let swapped = true;
    while (swapped) {
        swapped = false;
        for (let i=0; i < arr.length - 1; i++) {
            if (arr[i] > arr[i+1] ) {
                let temp = arr[i]
                arr[i] = arr[i+1]
                arr[i+1] = temp;
                swapped = true;
            }
        }
    }
    return arr;
};

console.log(bubbleSort([3,4,2,1,7]));